tfsdk/block.go

```diff
@@ -0,0 +1,412 @@
+package tfsdk
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+// Block defines the constraints and behaviors of a single structural field in a
+// schema.
+type Block struct {
+	tftypes.AttributePathStepper
```

Comment:
Does this work? What does it do, to embed an interface into a struct? 🤔

Reply:
Good catch. I thought it added compile-time verification that the interface was actually satisfied by the implementation, but It certainly does not. I'll replace it with the `var` approach. 👍